pkg/providers/types/interfaces.go

```diff
@@ -29,6 +30,15 @@ type ImageRead interface {
 	GetImage(ctx context.Context, organizationID, imageID string) (*Image, error)
 }
 
+type ImageWrite interface {
+	// CreateImageForUpload creates a new image resource for upload.
+	CreateImageForUpload(ctx context.Context, image *Image) (*Image, error)
+	// UploadImage uploads data to an image.
+	UploadImageData(ctx context.Context, imageID string, reader io.Reader) error
+	// DeleteImage deletes an image.
+	DeleteImage(ctx context.Context, imageID string) error
```

Comment:
Possibly this (and UploadImageData) should get an organizationID parameter, so they can ensure that the image in question does belong to the organization. That should be checked here or in the handler client, at least.